projects/hipdnn/plugins/miopen_legacy_plugin/integration_tests/IntegrationTestUtils.hpp

```
@@ -0,0 +1,204 @@
+        const std::array<const char*, 1> paths = {pluginPathStr.c_str()};
+        ASSERT_EQ(hipdnnSetEnginePluginPaths_ext(
+                      paths.size(), paths.data(), HIPDNN_PLUGIN_LOADING_ABSOLUTE),
+                  HIPDNN_STATUS_SUCCESS);
+
+        // Create handle and stream
+        ASSERT_EQ(hipdnnCreate(&_handle), HIPDNN_STATUS_SUCCESS);
+        ASSERT_EQ(hipStreamCreate(&_stream), hipSuccess);
+        ASSERT_EQ(hipdnnSetStream(_handle, _stream), HIPDNN_STATUS_SUCCESS);
+    }
+
+    void TearDown() override
+    {
+        if(_handle != nullptr)
+        {
+            ASSERT_EQ(hipdnnDestroy(_handle), HIPDNN_STATUS_SUCCESS);
+        }
+        if(_stream != nullptr)
+        {
+            ASSERT_EQ(hipStreamDestroy(_stream), hipSuccess);
+        }
+    }
+
+    virtual void runGraphTest(DataType tolerance, const TensorLayout& layout = TensorLayout::NCHW)
+        = 0;
+
+protected:
+    void verifyGraph(hipdnn_frontend::graph::Graph& graph,
+                     unsigned int seed,
+                     const IReferenceValidation& validator)
+    {
+        GraphTensorBundle gpuBundle = generateBundle(graph);
+        GraphTensorBundle cpuBundle = generateBundle(graph);
+
+        verifyGraph(graph, seed, cpuBundle, gpuBundle, validator);
+    }
+
+    void verifyGraph(hipdnn_frontend::graph::Graph& graph,
+                     unsigned int seed,
+                     GraphTensorBundle& cpuBundle,
+                     GraphTensorBundle& gpuBundle,
+                     const IReferenceValidation& validator)
+    {
+        initializeBundle(graph, gpuBundle, seed);
+        initializeBundle(graph, cpuBundle, seed);
+
+        auto result = graph.validate();
+        ASSERT_EQ(result.code, hipdnn_frontend::ErrorCode::OK) << result.err_msg;
+
+        executeGpuGraph(_handle, graph, gpuBundle);
+        executeCpuGraph(graph, cpuBundle);
+
+        for(const auto& tensorId : getAllNonVirtualOutputTensorIds(graph))
```

if we find zero tensors to validate, should we fail since thats unexpected?